File: app/src/main/java/nl/rsdt/japp/jotial/net/DownloadDrawableTask.java
package nl.rsdt.japp.jotial.net;

import android.graphics.drawable.Drawable;
import android.os.AsyncTask;
import android.util.Log;

import java.io.InputStream;
import java.net.URL;
import java.util.ArrayList;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 14-8-2016
 * Description...
 */
public class DownloadDrawableTask extends AsyncTask<URL, Integer, ArrayList<Drawable>> {

    private OnDowloadDrawablesCompletedCallback callback;

    public DownloadDrawableTask(OnDowloadDrawablesCompletedCallback callback) {
        this.callback = callback;
    }

    @Override
    protected ArrayList<Drawable> doInBackground(URL... params) {
        ArrayList<Drawable> drawables = new ArrayList<>();
        URL url;
        for(int i = 0; i < params.length; i++) {
            url = params[i];
            if(url != null) {
                try {
                    InputStream is = (InputStream)url.getContent();
                    drawables.add(Drawable.createFromStream(is, url.getFile()));
                }
                catch (Exception e) {
                    Log.e("UserInfo", e.toString(), e);
                }
            }
        }
        return drawables;
    }

    @Override
    protected void onPostExecute(ArrayList<Drawable> drawables) {
        super.onPostExecute(drawables);

        if(callback != null) {
            callback.onDownloadDrawablesCompleted(drawables);
        }
    }

    public interface OnDowloadDrawablesCompletedCallback {
        void onDownloadDrawablesCompleted(ArrayList<Drawable> drawables);
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/net/DownloadDrawableTask.kt
package nl.rsdt.japp.jotial.net

import android.graphics.drawable.Drawable
import android.os.AsyncTask
import android.util.Log
import java.io.InputStream
import java.net.URL
import java.util.*

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 14-8-2016
 * Description...
 */
class DownloadDrawableTask(private val callback: OnDowloadDrawablesCompletedCallback?) : AsyncTask<URL, Int, ArrayList<Drawable>>() {

    override fun doInBackground(vararg params: URL): ArrayList<Drawable> {
        val drawables = ArrayList<Drawable>()
        var url: URL?
        for (i in params.indices) {
            url = params[i]
            if (url != null) {
                try {
                    val `is` = url.content as InputStream
                    drawables.add(Drawable.createFromStream(`is`, url.file)!!)
                } catch (e: Exception) {
                    Log.e("UserInfo", e.toString(), e)
                }

            }
        }
        return drawables
    }

    override fun onPostExecute(drawables: ArrayList<Drawable>) {
        super.onPostExecute(drawables)

        callback?.onDownloadDrawablesCompleted(drawables)
    }

    interface OnDowloadDrawablesCompletedCallback {
        fun onDownloadDrawablesCompleted(drawables: ArrayList<Drawable>)
    }
}
